src/main/java/org/apache/commons/lang3/reflect/TypeUtils.java

```diff
@@ -1822,14 +1840,17 @@ private static void appendRecursiveTypes(final StringBuilder buf, final int[] re
             appendAllTo(buf.append('<'), ", ", argumentsFiltered).append('>');
         }
     }
-
-    private static int[] findRecursiveTypes(final ParameterizedType p) {
-        final Type[] filteredArgumentTypes = Arrays.copyOf(p.getActualTypeArguments(), p.getActualTypeArguments().length);
-        int[] indexesToRemove = {};
+    @SuppressWarnings("assignment.type.incompatible")/*
+    #3 Array.copyOf(array, n) returns an array of length n 
+    #4 Keeps on adding an element every time this statement is carried out, hence no. of elements <= no. of times loop runs, hence @LTEqLengthOf("p.getActualTypeArguments()")
+    */
+    private static int @LTEqLengthOf("#1.getActualTypeArguments()") [] findRecursiveTypes(final ParameterizedType p) {
+        final Type @SameLen("p.getActualTypeArguments()") [] filteredArgumentTypes = Arrays.copyOf(p.getActualTypeArguments(), p.getActualTypeArguments().length); // #3
+        int @LTEqLengthOf("p.getActualTypeArguments()") [] indexesToRemove = {};
```

Comment:
The `LTEqLengthOf` annotation is meant to be applied to integers that are less than or equal to the length of an array. Unfortunately, I think we currently don't have an annotation that expresses that an array is shorther or same length as another array.